### PR TITLE
Add tests for Timer and TimerFactory

### PR DESCRIPTION
**Pull request by AlexandraMash**
### Description

Added tests for Timer and TimerFactory

### API Changes
 
 None

### Platforms Affected

- Core (all platforms)

### Behavioral/Visual Changes

None

### Before/After Screenshots

Not applicable

### PR Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/Softeq/XToolkit.WhiteLabel/blob/master/.github/CONTRIBUTING.md) document
- [x] My code follows the [code styles](https://github.com/Softeq/dotnet-guidelines)
- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Review by SavikPavel — APPROVED** [review]

File: Softeq.XToolkit.Common.Tests/Timers/TimerTests/TimerTests.cs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+            var interval = 100;
+
+            var timer = new Timer(null, interval);
+            timer.Start();
+
+            Assert.True(timer.IsActive);
+
+            timer.Stop();
+        }
+
```
Comment (wcoder) [inline Softeq.XToolkit.Common.Tests/Timers/TimerTests/TimerTests.cs:53]: need case: double call `Start()`

File: Softeq.XToolkit.Common.Tests/Timers/TestFunctionClass.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+using System.Threading.Tasks;
+
+namespace Softeq.XToolkit.Common.Tests.Timers
+{
+    public class TestFunctionClass
+    {
+        public int ActionCount { get; set; }
+
+        public Task Function()
```
Comment (wcoder) [inline Softeq.XToolkit.Common.Tests/Timers/TestFunctionClass.cs:12]: seems like this class can be replaced via mock object

File: Softeq.XToolkit.Common/Timers/Timer.cs [hunk 3]
```diff
@@ -13,20 +13,27 @@ namespace Softeq.XToolkit.Common.Timers
...
-            _taskReference = taskReference;
-            _interval = interval;
+            TaskReference = taskReference;
+            Interval = interval;
         }
 
+        /// <summary>
+        /// Gets an interval setting of this timer.
+        /// </summary>
+        public int Interval { get; }
```
Comment (wcoder) [inline Softeq.XToolkit.Common/Timers/Timer.cs:30]: I think no need to expose these properties, it's out of the scope of current class

File: Softeq.XToolkit.Common/Timers/Timer.cs [hunk 4]
```diff
@@ -13,20 +13,27 @@ namespace Softeq.XToolkit.Common.Timers
...
-        private readonly int _interval;
-        private TaskReference? _taskReference;
-
         /// <summary>
         ///     Initializes a new instance of the <see cref="Timer"/> class with specified interval.
         /// </summary>
         /// <param name="taskReference">Task to be executed at specified interval.</param>
         /// <param name="interval">Timer interval (ms).</param>
         public Timer(TaskReference taskReference, int interval)
         {
```
Comment (wcoder) [inline Softeq.XToolkit.Common/Timers/Timer.cs:22]: can be added optional `ILogger` property to use in `FireAndForget`

File: Softeq.XToolkit.Common.Tests/Timers/TimerTests/TimerTests.cs [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+            functionClass.ActionCount = 0;
+            timer.Start();
+            await Task.Delay(delay);
+
+            Assert.True(timer.IsActive);
+            Assert.True(functionClass.ActionCount > 0, "Expected function to be called at least once");
+
+            timer.Stop();
+        }
+
```
Comment (wcoder) [inline Softeq.XToolkit.Common.Tests/Timers/TimerTests/TimerTests.cs:92]: need case: Start_After_Dispose

File: Softeq.XToolkit.Common.Tests/Timers/TimerTests/TimerTests.cs [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+            Assert.False(timer.IsActive);
+            Assert.NotNull(timer.TaskReference);
+        }
+
+        [Fact]
+        public async Task Stop_AfterStart_IsNotActive()
+        {
+            var functionClass = new TestFunctionClass();
+            var taskRef = new TaskReference(functionClass.Function);
```
Comment (wcoder) [inline Softeq.XToolkit.Common.Tests/Timers/TimerTests/TimerTests.cs:111]: `taskRef` can be shared

**Review by wcoder — CHANGES_REQUESTED** [review]

**Review by wcoder — APPROVED** [review]

**Review by SavikPavel — APPROVED** [review]

**Pull request merged**